lib/pronto/bundler_audit.rb

```diff
@@ -8,16 +8,40 @@ module Pronto
   # Pronto::BundlerAudit is a Pronto::Runner that:
   # 1. Updates the Ruby Advisory Database,
   # 2. Runs bundle-audit to scan the Gemfile.lock, and then
-  # 4. Returns an Array of Pronto::Message objects if any issues or advisories
+  # 3. Returns an Array of Pronto::Message objects if any issues or advisories
   # are found.
   class BundlerAudit < Runner
     GEMFILE_LOCK_FILENAME = "Gemfile.lock"
 
     # @return [Array<Pronto::Message>] per Pronto expectation
     def run
-      auditor = Auditor.new
-      auditor.call
+      return [] unless (results = Auditor.new.call) && results.size > 0
+
+      results.map do |result|
```

Comment:
This is all too much logic for one method for me. I'd like to see a MessageComposer object or something to deal with the results from Auditor.new.call. That object can return an empty array if no results are found and, otherwise compose them as a list of Messages. This should simplify testing as it reduces the number of conditions/responsibilities in one method.